src/main/java/com/snowflake/kafka/connector/internal/streaming/SchematizationUtils.java

```diff
@@ -154,7 +154,8 @@ private static Map<String, String> parseColumnTypes(JsonNode recordNode, Set<Str
     while (fields.hasNext()) {
       Map.Entry<String, JsonNode> field = fields.next();
       String colName = Utils.quoteNameIfNeeded(field.getKey());
-      if (columnNamesSet.contains(field.getKey())) {
+
+      if (columnNamesSet.contains(colName)) {
```

Comment:
cool so this can be reverted due to the 418 change in recordservice, this is the type of bug you can get when merging master into a fork! 😭